lib/colorls/flags.rb

```diff
@@ -37,12 +37,16 @@ def process
       # initialize locale from environment
       CLocale.setlocale(CLocale::LC_COLLATE, '')
 
-      return Core.new(@opts).ls if @args.empty?
+      @args = [Dir.pwd] if @args.empty?
       @args.sort!.each_with_index do |path, i|
-        next STDERR.puts "\n   Specified path '#{path}' doesn't exist.".colorize(:red) unless File.exist?(path)
-        puts '' if i > 0
-        puts "#{path}:" if Dir.exist?(path) && @args.size > 1
-        Core.new(path, @opts).ls
+        begin
+          next STDERR.puts "\n   Specified path '#{path}' doesn't exist.".colorize(:red) unless File.exist?(path)
+          puts '' if i > 0
+          puts "\n#{path}:" if Dir.exist?(path) && @args.size > 1
+          Core.new(path, @opts).ls
+        rescue StandardError => e
+          STDERR.puts "#{path}: #{e}"
```

Comment:
Why describe the `path` in the rescue part? It's not compulsory - an error could be even due to an incorrect flag right?

```ruby
STDERR.puts("colorls: #{e}".colorize(:red))
```

Reply:
I am not sure the error message from the exception always includes the current path, so the user would not know which of her arguments caused the error...